src/Planners/include/BipedalLocomotion/Planners/SwingFootPlanner.h

```diff
@@ -0,0 +1,100 @@
+/**
+ * @file SwingFootPlanner.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_PLANNERS_SWING_FOOT_PLANNER_H
+#define BIPEDAL_LOCOMOTION_PLANNERS_SWING_FOOT_PLANNER_H
+
+#include <memory>
+
+#include <manif/manif.h>
+
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/Planners/ContactList.h>
+#include <BipedalLocomotion/Planners/QuinticSpline.h>
+#include <BipedalLocomotion/Planners/SO3Planner.h>
+#include <BipedalLocomotion/System/Advanceable.h>
+
+namespace BipedalLocomotion
+{
+namespace Planners
+{
+
+/**
+ * SwingFootPlannerState describes the state of Swing foot planner.
+ */
+struct SwingFootPlannerState
+{
+    manif::SE3d transform; /**< Homogeneous transform between the link and the inertial frame */
+    manif::SE3d::Tangent spatialVelocity; /**< Spatial velocity written in mixed representation */
+    manif::SE3d::Tangent spatialAcceleration; /**< Spatial acceleration written in mixed
+                                                 representation */
```

Comment:
I would usually associate anything spatial to a right trivialized representation. In that case, it would be better to explicitly call these variables as `mixedVelocity` and `mixedAcceleration`, or `mixedTrivializedVelocity` and `mixedTrivializedAcceleration`.

@traversaro what's your take on this?

Reply:
I think that `mixedVelocity` and `mixedAcceleration` are fine. @traversaro what do you think?

Reply:
Yes, I agree. In most of the literature on rigid body dynamics, the "spatial" adjective is typically used in contrapposition w.r.t. to "classical" that is the mixed one.

Reply:
Done https://github.com/dic-iit/bipedal-locomotion-framework/pull/90/commits/b81089be999fcf85e6495715fad1dd7fce83c56c